src/v/lsm/core/internal/iterator.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2025 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "bytes/iobuf.h"
+#include "lsm/core/internal/keys.h"
+
+#include <seastar/core/future.hh>
+
+namespace lsm::internal {
+
+// A bi-directional iterator over sorted key-value pairs.
```

Comment:
Is it going to be the case that the way to have snapshot consistency across multiple reads is to reuse the same iterator? This seems to be the case from the tests, given we can call the seek* methods multiple times per iterator -- if so, maybe add it as a comment too? (though maybe there's a better place, or an already existing comment I haven't gotten to)